lib/ansible/config/base.yml

```diff
@@ -96,39 +96,6 @@ ANSIBLE_PIPELINING:
     key: pipelining
   type: boolean
   yaml: {key: plugins.connection.pipelining}
-ANSIBLE_SSH_ARGS:
-  # TODO: move to ssh plugin
-  default: -C -o ControlMaster=auto -o ControlPersist=60s
-  description:
-    - If set, this will override the Ansible default ssh arguments.
-    - In particular, users may wish to raise the ControlPersist time to encourage performance.  A value of 30 minutes may be appropriate.
-    - Be aware that if `-o ControlPath` is set in ssh_args, the control path setting is not used.
-  env: [{name: ANSIBLE_SSH_ARGS}]
-  ini:
-  - {key: ssh_args, section: ssh_connection}
-  yaml: {key: ssh_connection.ssh_args}
-ANSIBLE_SSH_CONTROL_PATH:
-  # TODO: move to ssh plugin
-  default: null
-  description:
-    - This is the location to save ssh's ControlPath sockets, it uses ssh's variable substitution.
-    - Since 2.3, if null, ansible will generate a unique hash. Use `%(directory)s` to indicate where to use the control dir path setting.
-    - Before 2.3 it defaulted to `control_path=%(directory)s/ansible-ssh-%%h-%%p-%%r`.
-    - Be aware that this setting is ignored if `-o ControlPath` is set in ssh args.
-  env: [{name: ANSIBLE_SSH_CONTROL_PATH}]
-  ini:
-  - {key: control_path, section: ssh_connection}
-  yaml: {key: ssh_connection.control_path}
-ANSIBLE_SSH_CONTROL_PATH_DIR:
-  # TODO: move to ssh plugin
-  default: ~/.ansible/cp
-  description:
-    - This sets the directory to use for ssh control path if the control path setting is null.
-    - Also, provides the `%(directory)s` variable for the control path setting.
-  env: [{name: ANSIBLE_SSH_CONTROL_PATH_DIR}]
-  ini:
-  - {key: control_path_dir, section: ssh_connection}
-  yaml: {key: ssh_connection.control_path_dir}
 ANSIBLE_SSH_EXECUTABLE:
```

Comment:
why leave this one here?

Reply:
[`C.ANSIBLE_SSH_EXECUTABLE`](https://github.com/ansible/ansible/blob/d37bd1f73b496b337dff77230fe1db63a93eadc6/lib/ansible/playbook/play_context.py#L147) is used before connection plugin instantiation, in order to [choose](https://github.com/ansible/ansible/blob/d37bd1f73b496b337dff77230fe1db63a93eadc6/lib/ansible/playbook/play_context.py#L588) which connection plugin to use.

Reply:
that we need to fix